### PR TITLE
Dev update install instructions

### LINKED ISSUE #109: Install soliket on macos M1
This is a compilation of instructions I followed to make SOLikeT work (and pass the tox tests) on the new macos M1.

System: Monterey 12.6.3, M1 chip, arm64

The main issue was to have tensorflow correctly installed (see [here](https://developer.apple.com/metal/tensorflow-plugin/)) and have cosmopower find it. 

1. Download latest miniconda installer (e.g., here: [Download Conda environment](https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-arm64.sh)) and properly rename it (e.g., -> miniconda.sh) 
2. Install miniconda and tensor flow-deps
`bash ~/miniconda.sh -b -p $HOME/miniconda`
`source ~/miniconda/bin/activate`
`conda install -c apple tensorflow-deps`
3. Create your virtual env
`conda env create -n my_env -f soliket-tests.yml`
`conda activate my_env`
4. Install tensorflow-macos and metal with correct versioning (I had them work with 2.9 and 0.5)
`pip install tensorflow-macos==2.9`
`pip install tensorflow-metal==0.5.0`
5. Download cosmopower manually
`git clone https://github.com/alessiospuriomancini/cosmopower`
`cd cosmopower`
6. Open requirements.txt and change the tensorflow dep as follows:
tensorflow>2.0 -> tensorflow-macos>2.0
7. Install cosmopower manually
`pip install .`
8. Go back to soliket folder and install it
`cd path/to/your/soliket`
`pip install -e .`

The above should let you install SOLikeT. 
In order to pass tests (both with pytest and within the tox env), I needed to add camb to soliket-tests.yml in the conda requirements to be installed via conda-forge (might work also via pip, not tried). Simply, add camb before the pip requirements.

### PR DESCRIPTION
**Pull request by mgerbino**
This PR updates install instructions including guidance on how to solve installation issues of cosmopower/tensorflow on M1 Mac. Fixes #109 

Still to be checked: install instructions at NERSC (beyond CORI)

### CODE REVIEW

**Comment by itrharrison** [comment]
@HTJense is the [installation of the conda env](https://github.com/simonsobs/SOLikeT/blob/ff2f60bfcc2932dfc016583ca12f291874209d10/.readthedocs.yml#L14) necessary for readthedocs? It slows down the docs build quite a bit.

**Comment by HTJense** [comment]
> @HTJense is the [installation of the conda env](https://github.com/simonsobs/SOLikeT/blob/ff2f60bfcc2932dfc016583ca12f291874209d10/.readthedocs.yml#L14) necessary for readthedocs? It slows down the docs build quite a bit.

I don't know, I doubt it, but you can try without it and see if it works better 🤷🏾‍♀️

File: docs/developers.rst [hunk 1]
```diff
@@ -2,206 +2,214 @@
...
+----------------------------------
+Check SOLikeT is working as intended in a python environment of your own specification (i.e. you have installed SOLikeT not using the soliket-tests conda environment).
+
+For this you need to make sure all of the required system-level and python dependencies described in `the installation instructions <install.html>`_ are working correctly, then run::
+
+  pytest -v soliket
+
+Good luck!
+
+If your unit tests check the statistical distribution of a random sample, the test outcome itself is a random variable, and the test will fail from time to time. Please mark such tests with the ``@pytest.mark.flaky`` decorator, so that they will be automatically tried again on failure. To prevent non-random test failures from being run multiple times, please isolate random statistical tests and deterministic tests in their own test cases.
```
Comment (reneehlozek) [inline docs/developers.rst:206]: The last discussion on unit tests seems too detailed/technical. Should we put this in parenthesis - I'm worried that a significant number of people will be confused by it

File: docs/documentation.rst [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+add multiple ``autoclass`` directives, one for each class you have. The
+``:members:`` directive will cause sphinx to turn the docstrings for each of the
+class's member functions to be turned into an entry.
+
+Sometimes, your class might inherit a lot of functions from some parent class,
+or implement a lot of boilerplate functions that are needed to interface with
+some other codes or frameworks. These kinds of functions can at times give
+excessive documentation that the end-user might not really need or should not
+care about. In our Pulsars example, we had the ``extra_function`` that we did
+not want to add to our docs page. For this, there exists the
```
Comment (reneehlozek) [inline docs/documentation.rst:131]: While this is useful information, we don't give a directive to make sure people keep this up to date. You could imagine a code that stops being extraneous but remains a ghost in the docpage?

File: docs/documentation.rst [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+``:exclude-members:`` directive, that will skip given functions that people
+might not need.
+
+
+Obviously, there are many many more ways in which you can make your documentation
```
Comment (reneehlozek) [inline docs/documentation.rst:136]: I would remove this first sentence, it's unnecessary

File: docs/workflow.rst [hunk 4]
```diff
@@ -11,7 +9,7 @@ If you are not a member of the Simons Observatory Collaboration with access to t
 If you are planning on bringing a new Likelihood in to SOLikeT, please have a look at the `guidelines <https://github.com/simonsobs/SOLikeT/blob/master/guidelines.md>`_ which explain how this should be done.
```
Comment (reneehlozek) [inline docs/workflow.rst:10]: Do we want to add a note to contact the developers of SOLikeT (or not???)

**Review by reneehlozek — APPROVED** [review]
Some general comments and suggested changes

**Review by reneehlozek — APPROVED** [review]
🍹

**Pull request merged**